extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/cloudtraillogs.go

```diff
@@ -0,0 +1,308 @@
+	EventID                      string         `json:"eventID"`
+	EventType                    string         `json:"eventType"`
+	EventCategory                string         `json:"eventCategory"`
+	RecipientAccountID           string         `json:"recipientAccountId"`
+	UserIdentity                 map[string]any `json:"userIdentity"`
+	ResponseElements             map[string]any `json:"responseElements"`
+	RequestParameters            map[string]any `json:"requestParameters"`
+	Resources                    []any          `json:"resources"`
+	ReadOnly                     bool           `json:"readOnly"`
+	ManagementEvent              bool           `json:"managementEvent"`
+	AdditionalEventData          map[string]any `json:"additionalEventData"`
+	TLSDetails                   map[string]any `json:"tlsDetails"`
+	SessionCredentialFromConsole string         `json:"sessionCredentialFromConsole"`
+}
+
+type CloudTrailLogs struct {
+	Records []CloudTrailRecord `json:"Records"`
+}
+
+func NewCloudTrailLogsUnmarshaler(buildInfo component.BuildInfo) plog.Unmarshaler {
+	return &CloudTrailLogsUnmarshaler{
+		buildInfo: buildInfo,
+		gzipPool:  sync.Pool{},
+	}
+}
+
+func (u *CloudTrailLogsUnmarshaler) Unmarshal(buf []byte) (plog.Logs, error) {
+	return u.UnmarshalLogs(buf)
+}
+
+func (u *CloudTrailLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
+	// Check if the content is gzip-compressed
+	var errDecompress error
+	gzipReader, ok := u.gzipPool.Get().(*gzip.Reader)
+	if !ok {
+		gzipReader, errDecompress = gzip.NewReader(bytes.NewReader(buf))
+	} else {
+		errDecompress = gzipReader.Reset(bytes.NewReader(buf))
+	}
+
+	var decompressedBuf []byte
+	if errDecompress == nil {
+		// Content is gzip-compressed, decompress it
+		defer func() {
+			_ = gzipReader.Close()
+			u.gzipPool.Put(gzipReader)
+		}()
+
+		decompressedBuf, errDecompress = io.ReadAll(gzipReader)
+		if errDecompress != nil {
+			return plog.Logs{}, fmt.Errorf("failed to decompress CloudTrail logs: %w", errDecompress)
+		}
+	} else {
+		// Content is not gzip-compressed, use the original buffer
+		decompressedBuf = buf
+		if gzipReader != nil {
+			u.gzipPool.Put(gzipReader)
+		}
+	}
+
+	var cloudTrailLogs CloudTrailLogs
+	if err := json.Unmarshal(decompressedBuf, &cloudTrailLogs); err != nil || len(cloudTrailLogs.Records) == 0 {
+		return plog.Logs{}, fmt.Errorf("failed to unmarshal CloudTrail logs: %w", err)
+	}
+
+	return u.processRecords(cloudTrailLogs.Records), nil
+}
+
+func (u *CloudTrailLogsUnmarshaler) processRecords(records []CloudTrailRecord) plog.Logs {
+	logs := plog.NewLogs()
+
+	// Group records by account ID and region to create separate resource logs
+	groupedRecords := make(map[string][]CloudTrailRecord)
+	for _, record := range records {
+		// Use account ID and region as the key for grouping
+		key := record.RecipientAccountID + "|" + record.AwsRegion
+		groupedRecords[key] = append(groupedRecords[key], record)
+	}
```

Comment:
From the [documentation](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-log-file-examples.html) it says:

> CloudTrail uses the following file name format for the log file objects that it delivers to your Amazon S3 bucket:
> AccountID_CloudTrail_RegionName_YYYYMMDDTHHmmZ_UniqueString.FileNameFormat 

In that case, would it be correct to assume all logs coming from the same S3 file (I assume this is the case) have the same account ID and region?